conda_forge_tick/migrators/libboost.py

```diff
@@ -12,44 +12,105 @@ def _slice_into_output_sections(meta_yaml_lines, attrs):
+    Takes a list of lines and returns a dict from each output index to
     the list of lines where this output is described in the meta.yaml.
-    The result will always contain a "global" section (== everything
-    if there are no other outputs).
+    The result will always contain an index -1 for the top-level section (
+    == everything if there are no other outputs).
     """
-    outputs = attrs["meta_yaml"].get("outputs", [])
-    output_names = [o["name"] for o in outputs]
-    # if there are no outputs, there's only one section
-    if not output_names:
-        return {"global": meta_yaml_lines}
-    # output_names may contain duplicates; remove them but keep order
-    names = []
-    [names := names + [x] for x in output_names if x not in names]
-    num_outputs = len(names)
-    # add dummy for later use & reverse list for easier pop()ing
-    names += ["dummy"]
-    names.reverse()
-    # initialize
-    pos, prev, seek = 0, "global", names.pop()
+    outputs_token_pos = None
+    re_output_start = None
+    re_outputs_token = re.compile(r"^\s*outputs:.*")
+    re_outputs_token_list = re.compile(r"^\s*outputs:\s*\[.*")
+    re_match_block_list = re.compile(r"^\s*\-.*")
+
+    pos = 0
+    section_index = -1
     sections = {}
     for i, line in enumerate(meta_yaml_lines):
-        # assumes order of names matches their appearance in meta_yaml,
-        # and that they appear literally (i.e. no {{...}}) and without quotes
-        if f"- name: {seek}" in line:
-            # found the beginning of the next output;
-            # everything until here belongs to the previous one
-            sections[prev] = meta_yaml_lines[pos:i]
-            # update
-            pos, prev, seek = i, seek, names.pop()
-            if seek == "dummy":
-                # reached the last output; it goes until the end of the file
-                sections[prev] = meta_yaml_lines[pos:]
-    if len(sections) != num_outputs + 1:
-        raise RuntimeError("Could not find all output sections in meta.yaml!")
-    return sections
-
-
-def _process_section(name, attrs, lines):
+        # we go through lines until we find the `outputs:` line
+        # we grab its position for later
+        if re_outputs_token.match(line) and outputs_token_pos is None:
+            # we cannot handle list-style outputs so raise an error
+            if re_outputs_token_list.match(line):
+                raise RuntimeError(
+                    "List-style outputs not supported for outputs slicing!"
+                )
+
+            outputs_token_pos = i
+            continue
+
+        # next we need to find the indent of the first output list element.
+        # it could be anything from the indent of the `outputs:` line to that number
+        # plus zero or more spaces. Typically, the indent is the same as the `outputs:`
+        # plus one of [0, 2, 4] spaces.
+        # ocne we find the first output list element, we build a regex to match any list
```

Comment:
ocne -> once

(though I'd like to avoid the indentation handling completely - I don't (yet?) see why it's necessary)

Reply:
Thanks. I will fix the typo. 

The indentation handling is because yaml allows any consistent indent below a given section. The outputs are a list, is you can have

"""
outputs:
- blah
"""

Or 

"""
outputs:
  - blah
"""

Or w/e

Reply:
Looking for list items at any indent is not robust ofc, so we have to find a list item at the correct indent. Even that is not robust since we don't know when the outputs section ends. Thus we clean that up later by requiring a `name: ` key in the output. This is a heuristic since in principle that string could appear in other parts of the recipe.

Reply:
We could detect the indent of the dict in each output, record it, and then find the right name key at the correct indent.

Reply:
Personally, I just want to not care about the indent at all here, and let the regex portion of `\s*` take care of it.

Reply:
That's less robust and leaves edge cases.